### PR TITLE
refactor: change r precision calculation and documentation

### PR DESCRIPTION
**Pull request by guenthermi**
Goals:

- correct the implementation of R-Precision
- add hints in the documentation about potentially incorrect evaluation scores
- add tests for all metric functions
- add links to developer reference of metric functions in the documentation

- [x] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

**Comment by guenthermi** [comment]
tests will be added in #617

File: docarray/math/evaluation.py [hunk 1]
```diff
@@ -12,20 +12,25 @@ def _check_k(k):
 
 
 def r_precision(binary_relevance: List[int], **kwargs) -> float:
```
Comment (JoanFM) [inline docarray/math/evaluation.py:14]: there needs to be some tests testing the new behavior

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docarray/math/evaluation.py (hunk 1 above)
Comment (guenthermi) [reply docarray/math/evaluation.py:14]: Yes I agree. Currently there are no tests for the metric functions. I have created a test which tests the behavior together with the evaluate function functions in #617 (which has to be adopted -after one of those Pars is merged), but I can also create tests which directly test the functions docarray.math here.

**Review by bwanglzu — COMMENTED** [review]
add tests otherwise looks good

File: docarray/math/evaluation.py (hunk 1 above)
Comment (JoanFM) [reply docarray/math/evaluation.py:14]: add unit tests yes

**Review by JoanFM — APPROVED** [review]

**Pull request merged**